  This patch prevents gettextize from screwing up any of the ChangeLogs
  or the makefiles or the configure files.  The 0.11.x releases are
  even worse than the 0.10.x releases :(
 
--- gettext-0.11.5/misc/gettextize.in.old	2002-08-05 19:51:35.000000000 +0800
+++ gettext-0.11.5/misc/gettextize.in	2002-12-17 23:10:29.000000000 +0800
@@ -262,44 +262,12 @@
 # manage the ChangeLog file, relative to $srcdir.
 func_ChangeLog_init ()
 {
-  modified_ChangeLog=
 }
 func_ChangeLog_add_entry ()
 {
-  if $doit; then
-    if test -z "$modified_ChangeLog"; then
-      echo "$date  gettextize  <bug-gnu-gettext@gnu.org>" > "$srcdir/ChangeLog.tmp"
-      echo >> "$srcdir/ChangeLog.tmp"
-      modified_ChangeLog=yes
-    fi
-    echo "$1" >> "$srcdir/ChangeLog.tmp"
-  else
-    modified_ChangeLog=yes
-  fi
 }
 func_ChangeLog_finish ()
 {
-  if test -n "$modified_ChangeLog"; then
-    if $doit; then
-      echo >> "$srcdir/ChangeLog.tmp"
-      if test -f "$srcdir/ChangeLog"; then
-        echo "Adding an entry to ChangeLog (backup is in ChangeLog~)"
-        cat "$srcdir/ChangeLog" >> "$srcdir/ChangeLog.tmp"
-        rm -f "$srcdir/ChangeLog~"
-        cp -p "$srcdir/ChangeLog" "$srcdir/ChangeLog~"
-      else
-        echo "Creating ChangeLog"
-      fi
-      cp "$srcdir/ChangeLog.tmp" "$srcdir/ChangeLog"
-      rm -f "$srcdir/ChangeLog.tmp"
-    else
-      if test -f "$srcdir/ChangeLog"; then
-        echo "Add an entry to ChangeLog"
-      else
-        echo "Create ChangeLog"
-      fi
-    fi
-  fi
 }
 
 # func_poChangeLog_init
@@ -308,44 +276,12 @@
 # manage the po/ChangeLog file, relative to $srcdir.
 func_poChangeLog_init ()
 {
-  modified_poChangeLog=
 }
 func_poChangeLog_add_entry ()
 {
-  if $doit; then
-    if test -z "$modified_poChangeLog"; then
-      echo "$date  gettextize  <bug-gnu-gettext@gnu.org>" > "$srcdir/po/ChangeLog.tmp"
-      echo >> "$srcdir/po/ChangeLog.tmp"
-      modified_poChangeLog=yes
-    fi
-    echo "$1" >> "$srcdir/po/ChangeLog.tmp"
-  else
-    modified_poChangeLog=yes
-  fi
 }
 func_poChangeLog_finish ()
 {
-  if test -n "$modified_poChangeLog"; then
-    if $doit; then
-      echo >> "$srcdir/po/ChangeLog.tmp"
-      if test -f "$srcdir/po/ChangeLog"; then
-        echo "Adding an entry to po/ChangeLog (backup is in po/ChangeLog~)"
-        cat "$srcdir/po/ChangeLog" >> "$srcdir/po/ChangeLog.tmp"
-        rm -f "$srcdir/po/ChangeLog~"
-        cp -p "$srcdir/po/ChangeLog" "$srcdir/po/ChangeLog~"
-      else
-        echo "Creating po/ChangeLog"
-      fi
-      cp "$srcdir/po/ChangeLog.tmp" "$srcdir/po/ChangeLog"
-      rm -f "$srcdir/po/ChangeLog.tmp"
-    else
-      if test -f "$srcdir/po/ChangeLog"; then
-        echo "Add an entry to po/ChangeLog"
-      else
-        echo "Create po/ChangeLog"
-      fi
-    fi
-  fi
 }
 
 # func_m4ChangeLog_init
@@ -354,44 +290,12 @@
 # manage the $m4dir/ChangeLog file, relative to $srcdir.
 func_m4ChangeLog_init ()
 {
-  modified_m4ChangeLog=
 }
 func_m4ChangeLog_add_entry ()
 {
-  if $doit; then
-    if test -z "$modified_m4ChangeLog"; then
-      echo "$date  gettextize  <bug-gnu-gettext@gnu.org>" > "$srcdir/$m4dir/ChangeLog.tmp"
-      echo >> "$srcdir/$m4dir/ChangeLog.tmp"
-      modified_m4ChangeLog=yes
-    fi
-    echo "$1" >> "$srcdir/$m4dir/ChangeLog.tmp"
-  else
-    modified_m4ChangeLog=yes
-  fi
 }
 func_m4ChangeLog_finish ()
 {
-  if test -n "$modified_m4ChangeLog"; then
-    if $doit; then
-      echo >> "$srcdir/$m4dir/ChangeLog.tmp"
-      if test -f "$srcdir/$m4dir/ChangeLog"; then
-        echo "Adding an entry to $m4dir/ChangeLog (backup is in $m4dir/ChangeLog~)"
-        cat "$srcdir/$m4dir/ChangeLog" >> "$srcdir/$m4dir/ChangeLog.tmp"
-        rm -f "$srcdir/$m4dir/ChangeLog~"
-        cp -p "$srcdir/$m4dir/ChangeLog" "$srcdir/$m4dir/ChangeLog~"
-      else
-        echo "Creating $m4dir/ChangeLog"
-      fi
-      cp "$srcdir/$m4dir/ChangeLog.tmp" "$srcdir/$m4dir/ChangeLog"
-      rm -f "$srcdir/$m4dir/ChangeLog.tmp"
-    else
-      if test -f "$srcdir/$m4dir/ChangeLog"; then
-        echo "Add an entry to $m4dir/ChangeLog"
-      else
-        echo "Create $m4dir/ChangeLog"
-      fi
-    fi
-  fi
 }
 
 if test ! -f "$srcdir/intl/Makefile.in" && test -n "$intldir"; then
@@ -559,320 +463,6 @@
 fi
 $do_changelog && func_poChangeLog_finish
 
-m4filelist='  codeset.m4 gettext.m4 glibc21.m4 iconv.m4 intdiv0.m4 inttypes.m4
-  inttypes_h.m4 inttypes-pri.m4 isc-posix.m4 lcmessage.m4 lib-ld.m4 lib-link.m4
-  lib-prefix.m4 progtest.m4 stdint_h.m4 uintmax_t.m4 ulonglong.m4'
-# We cannot omit codeset.m4, glibc21.m4, intdiv0.m4, inttypes.m4,
-# inttypes_h.m4, isc-posix.m4, lcmessage.m4, stdint_h.m4, uintmax_t.m4
-# if test -z "$intldir", otherwise "aclocal -I m4" might give an error.
-# (aclocal doesn't know which macros are really needed, it looks which macros
-# are potentially needed.)
-
-# All sorts of bugs could occur if the configure file was remade with the wrong
-# version of gettext.m4 et al. (because then the configure and the po/Makefile.in.in
-# don't fit together). It is therefore important that the package carries the
-# right versions of gettext.m4 et al. with it.
-if test -f "$srcdir/Makefile.am"; then
-  # A package using automake.
-
-  # Extract the macro directory name from Makefile.am.
-  aclocal_amflags=`grep '^ACLOCAL_AMFLAGS[ 	]*=' "$srcdir/Makefile.am" | sed -e 's/^ACLOCAL_AMFLAGS[ 	]*=\(.*\)$/\1/'`
-  m4dir=m4
-  m4dir_is_next=
-  for arg in $aclocal_amflags; do
-    if test -n "$m4dir_is_next"; then
-      m4dir="$arg"
-      break
-    else
-      if test "X$arg" = "X-I"; then
-        m4dir_is_next=yes
-      else
-        m4dir_is_next=
-      fi
-    fi
-  done
-
-  # Update the *.m4 files and the corresponding Makefile.am.
-  $do_changelog && func_m4ChangeLog_init
-  added_m4files=
-  if test -d "$srcdir/$m4dir"; then
-    :
-  else
-    if $doit; then
-      echo "Creating directory $m4dir"
-      mkdir "$srcdir/$m4dir"
-    else
-      echo "Create directory $m4dir"
-    fi
-    added_directories="$added_directories $m4dir"
-  fi
-  for file in $m4filelist; do
-    same=no
-    if test -f "$srcdir/$m4dir/$file"; then
-      if cmp -s "@datadir@/aclocal/$file" "$srcdir/$m4dir/$file"; then
-        same=yes
-      fi
-    else
-      added_m4files="$added_m4files $file"
-    fi
-    if $do_changelog && test $same = no; then
-      if test -f "$srcdir/$m4dir/$file"; then
-        func_m4ChangeLog_add_entry "	* $file: Upgrade to gettext-${version}."
-      else
-        func_m4ChangeLog_add_entry "	* $file: New file, from gettext-${version}."
-      fi
-    fi
-    func_backup "$m4dir/$file"
-    func_linkorcopy "@datadir@/aclocal/$file" "@datadir@/aclocal/$file" "$m4dir/$file"
-  done
-  if test -n "$added_m4files"; then
-    if test -f "$srcdir/$m4dir/Makefile.am"; then
-      if $doit; then
-        echo "Updating EXTRA_DIST in $m4dir/Makefile.am (backup is in $m4dir/Makefile.am~)"
-        func_backup "$m4dir/Makefile.am"
-        rm -f "$srcdir/$m4dir/Makefile.am"
-        if grep '^EXTRA_DIST[ 	]*=' "$srcdir/$m4dir/Makefile.am~" > /dev/null; then
-          sed -e "s%^\(EXTRA_DIST[ 	]*=\)%\\1$added_m4files %" < "$srcdir/$m4dir/Makefile.am~" > "$srcdir/$m4dir/Makefile.am"
-          $do_changelog && func_m4ChangeLog_add_entry "	* Makefile.am (EXTRA_DIST): Add the new files."
-        else
-          (cat "$srcdir/$m4dir/Makefile.am~"; echo; echo "EXTRA_DIST =$added_m4files") > "$srcdir/$m4dir/Makefile.am"
-          $do_changelog && func_m4ChangeLog_add_entry "	* Makefile.am (EXTRA_DIST): New variable."
-        fi
-      else
-        echo "Update EXTRA_DIST in $m4dir/Makefile.am"
-        $do_changelog && func_m4ChangeLog_add_entry "	* Makefile.am (EXTRA_DIST)."
-      fi
-    else
-      if $doit; then
-        echo "Creating $m4dir/Makefile.am"
-        echo "EXTRA_DIST =$added_m4files" > "$srcdir/$m4dir/Makefile.am"
-      else
-        echo "Create $m4dir/Makefile.am"
-      fi
-      $do_changelog && func_m4ChangeLog_add_entry "	* Makefile.am: New file."
-      added_acoutput="$added_acoutput $m4dir/Makefile"
-    fi
-  fi
-  $do_changelog && func_m4ChangeLog_finish
-  # Also create $m4dir/Makefile.in from $m4dir/Makefile.am, because automake
-  # doesn't do it by itself.
-  if $doit; then
-    case "$added_acoutput" in
-      *" $m4dir/Makefile")
-        (cd "$srcdir" && automake "$m4dir/Makefile") 2>/dev/null ||
-        please="$please
-Please run 'automake $m4dir/Makefile' to create $m4dir/Makefile.in
-"
-        ;;
-    esac
-  fi
-
-  # Update the top-level Makefile.am.
-  modified_Makefile_am=
-  # func_modify_Makefile_am changelog_comment
-  # assumes a modified copy of $srcdir/Makefile.am in $srcdir/Makefile.am.tmp
-  # and replaces the original Makefile.am file with the modified one if
-  # the two files differ. Then it removes the modified copy.
-  func_modify_Makefile_am ()
-  {
-    if cmp -s "$srcdir/Makefile.am" "$srcdir/Makefile.am.tmp"; then
-      :
-    else
-      if test -z "$modified_Makefile_am"; then
-        if $doit; then
-          echo "Updating Makefile.am (backup is in Makefile.am~)"
-          func_backup Makefile.am
-        else
-          echo "Update Makefile.am"
-        fi
-      fi
-      if $doit; then
-        rm -f "$srcdir/Makefile.am"
-        cp "$srcdir/Makefile.am.tmp" "$srcdir/Makefile.am"
-      fi
-      if $do_changelog; then
-        if test -z "$modified_Makefile_am"; then
-          func_ChangeLog_add_entry "	* Makefile.am $1"
-        else
-          func_ChangeLog_add_entry "	$1"
-        fi
-      fi
-      modified_Makefile_am=yes
-    fi
-    rm -f "$srcdir/Makefile.am.tmp"
-  }
-
-  if test -n "$added_directories"; then
-    if grep '^SUBDIRS[ 	]*=' "$srcdir/Makefile.am" > /dev/null; then
-      sed -e "s%^\(SUBDIRS[ 	]*=\)%\\1$added_directories %" < "$srcdir/Makefile.am" > "$srcdir/Makefile.am.tmp"
-      func_modify_Makefile_am "(SUBDIRS): Add "`echo $added_directories | sed -e 's/ /, /g'`'.'
-    else
-      (cat "$srcdir/Makefile.am"; echo; echo "SUBDIRS =$added_directories") > "$srcdir/Makefile.am.tmp"
-      func_modify_Makefile_am "(SUBDIRS): New variable."
-    fi
-  fi
-  if test -n "$removed_directory"; then
-    sed -e '/^SUBDIRS[ 	]*=/ {
-        :a
-        s%\([ 	]\)'"$removed_directory"'[ 	]%\1%
-        s%[ 	]'"$removed_directory"'$%%
-        tb
-        :b
-        s%\\$%\\%
-        tc
-        bd
-        :c
-        n
-        ba
-      :d
-    }' < "$srcdir/Makefile.am" > "$srcdir/Makefile.am.tmp"
-    func_modify_Makefile_am "(SUBDIRS): Remove $removed_directory."
-  fi
-  if test -z "$m4dir_is_next"; then
-    if grep '^ACLOCAL_AMFLAGS[ 	]*=' "$srcdir/Makefile.am" > /dev/null; then
-      sed -e "s%^\(ACLOCAL_AMFLAGS[ 	]*=\)%\\1 -I $m4dir %" < "$srcdir/Makefile.am" > "$srcdir/Makefile.am.tmp"
-      func_modify_Makefile_am "(ACLOCAL_AMFLAGS): Add -I $m4dir."
-    else
-      (cat "$srcdir/Makefile.am"; echo; echo "ACLOCAL_AMFLAGS = -I $m4dir") > "$srcdir/Makefile.am.tmp"
-      func_modify_Makefile_am "(ACLOCAL_AMFLAGS): New variable."
-    fi
-    # Also update Makefile.in and, if existent, Makefile. Otherwise they
-    # would take into account the new flags only after a few rounds of
-    # "./configure", "make", "touch configure.in", "make distclean".
-    if $doit; then
-      for file in Makefile.in Makefile; do
-        if test -f "$srcdir/$file"; then
-          func_backup $file
-          rm -f "$srcdir/$file"
-          sed -e "s%(ACLOCAL)%(ACLOCAL) -I $m4dir%" < "$srcdir/$file~" > "$srcdir/$file"
-        fi
-      done
-    fi
-  fi
-  if test -n "$added_extradist"; then
-    if grep '^EXTRA_DIST[ 	]*=' "$srcdir/Makefile.am" > /dev/null; then
-      sed -e "s%^\(EXTRA_DIST[ 	]*=\)%\\1$added_extradist %" < "$srcdir/Makefile.am" > "$srcdir/Makefile.am.tmp"
-      func_modify_Makefile_am "(EXTRA_DIST): Add$added_extradist."
-    else
-      (cat "$srcdir/Makefile.am"; echo; echo "EXTRA_DIST =$added_extradist") > "$srcdir/Makefile.am.tmp"
-      func_modify_Makefile_am "(EXTRA_DIST): New variable."
-    fi
-  fi
-  please="$please
-Please run 'aclocal -I $m4dir' to regenerate the aclocal.m4 file.
-You need aclocal from GNU automake 1.5 (or newer) to do this.
-Then run 'autoconf' to regenerate the configure file.
-"
-else
-  please="$please
-Please add the files
-$m4filelist
-from the @datadir@/aclocal directory to your aclocal.m4 file.
-"
-fi
-
-modified_configure_in=
-# func_modify_configure_in changelog_comment
-# assumes a modified copy of $srcdir/$configure_in in $srcdir/$configure_in.tmp
-# and replaces the original configure.in/ac file with the modified one if
-# the two files differ. Then it removes the modified copy.
-func_modify_configure_in ()
-{
-  if cmp -s "$srcdir/$configure_in" "$srcdir/$configure_in.tmp"; then
-    :
-  else
-    if test -z "$modified_configure_in"; then
-      if $doit; then
-        echo "Updating $configure_in (backup is in $configure_in~)"
-        func_backup $configure_in
-      else
-        echo "Update $configure_in"
-      fi
-    fi
-    if $doit; then
-      rm -f "$srcdir/$configure_in"
-      cp "$srcdir/$configure_in.tmp" "$srcdir/$configure_in"
-    fi
-    if $do_changelog; then
-      if test -z "$modified_configure_in"; then
-        func_ChangeLog_add_entry "	* $configure_in $1"
-      else
-        func_ChangeLog_add_entry "	$1"
-      fi
-    fi
-    modified_configure_in=yes
-  fi
-  rm -f "$srcdir/$configure_in.tmp"
-}
-
-if test -n "$added_acoutput"; then
-  if grep '^AC_CONFIG_FILES(' "$srcdir/$configure_in" > /dev/null; then
-    sed -e "s%^\\(AC_CONFIG_FILES([^])\\,]*\\)%\\1$added_acoutput %" < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-    func_modify_configure_in "(AC_CONFIG_FILES): Add "`echo $added_acoutput | sed -e 's/ /, /g'`'.'
-  else
-    if grep '^AC_OUTPUT(' "$srcdir/$configure_in" > /dev/null; then
-      sed -e "s%^\\(AC_OUTPUT([^])\\,]*\\)%\\1$added_acoutput %" < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-      func_modify_configure_in "(AC_OUTPUT): Add "`echo $added_acoutput | sed -e 's/ /, /g'`'.'
-    else
-      please="$please
-Please add$added_acoutput to the AC_OUTPUT or AC_CONFIG_FILES invocation in the $configure_in file.
-"
-    fi
-  fi
-fi
-if test -n "$removed_acoutput"; then
-  for file in $removed_acoutput; do
-    tag=
-    sedprog='{
-      s%\([[ 	]\)'"$file"'[ 	]%\1%
-      s%\([[ 	]\)'"$file"'\([]),]\)%\1\2%
-      s%[[ 	]'"$file"'$%%
-        :a
-        tb
-        :b
-        s%\\$%\\%
-        tc
-        bd
-        :c
-        n
-        s%\([ 	]\)'"$file"'[ 	]%\1%
-        s%\([ 	]\)'"$file"'\([]),]\)%\1\2%
-        s%[ 	]'"$file"'$%%
-        ba
-      :d
-    }'
-    sed -e '/^AC_CONFIG_FILES(/'"$sedprog" < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-    if cmp -s "$srcdir/$configure_in" "$srcdir/$configure_in.tmp"; then
-      sed -e '/^AC_OUTPUT(/'"$sedprog" < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-      if cmp -s "$srcdir/$configure_in" "$srcdir/$configure_in.tmp"; then
-        :
-      else
-        tag=AC_OUTPUT
-      fi
-    else
-      tag=AC_CONFIG_FILES
-    fi
-    if test -n "$tag"; then
-      func_modify_configure_in "($tag): Remove $file."
-    else
-      rm -f "$srcdir/$configure_in.tmp"
-      if test "$file" != intl/intlh.inst; then
-        please="$please
-Please remove $file from the AC_OUTPUT or AC_CONFIG_FILES invocation
-in the $configure_in file.
-"
-      fi
-    fi
-  done
-fi
-sed -e 's%sed -e "/POTFILES =/r po/POTFILES" po/Makefile\.in > po/Makefile *;* *%%' < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-func_modify_configure_in "(AC_OUTPUT): Remove command that created po/Makefile."
-sed -e '/^\(dnl \|\)AC_LINK_FILES(\$nls_cv_header_libgt, \$nls_cv_header_intl)$/d' < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-func_modify_configure_in "(AC_LINK_FILES): Remove invocation."
-sed -e 's/^AM_GNU_GETTEXT_VERSION([^()]*)/AM_GNU_GETTEXT_VERSION('"$version"'/' < "$srcdir/$configure_in" > "$srcdir/$configure_in.tmp"
-func_modify_configure_in "(AM_GNU_GETTEXT_VERSION): Bump to $version."
-$do_changelog && func_ChangeLog_finish
-
 # Recommend replacement for deprecated Makefile variables.
 use_libtool=`cat "$srcdir/$configure_in" | grep '^A[CM]_PROG_LIBTOOL'`
 for file in `(cd "$srcdir"; find . -name Makefile.am -print; find . -name Makefile.in -print) | sed -e 's,^\./,,'`; do
